[IV-23-24]Objetivo 6

Christianlr opened the pull request:
* [x] ¿Se han establecido criterios *a priori* para elegir el sistema de
integración continua más conveniente y se han calificado los sistemas que
cumplan esos requisitos según esos criterios?
* [x] ¿Se han examinado varios sistemas de integración continua?
* [x] ¿Se han configurado varios sistemas de integración continua?
* [x] ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que
   sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar
   desde los tests)?
* [x] ¿Uno de los sistemas configurados permite comprobar cuales son las versiones
   dellenguaje con las que funciona correctamente nuestra aplicación?
* [x] ¿Se escogen de forma adecuada las versiones del lenguaje que se testean,
   tanto en el sistema de CI como en el contenedor Docker?
* [x] ¿Se han justificado correctamente las versiones del lenguaje que se están
testeando y se ha comprobado que no se comprueban varias veces lo mismo (la
misma versión en CI y en el contenedor Docker en otro sistema CI?

cloudbees-platform[bot] committed:
Creación workflow con Cloudbees close#59

cloudbees-platform[bot] committed:
Creación de workflow con Cloudbees close#59

cloudbees-platform[bot] committed:
Creación workflow con Cloudbees close #59

cloudbees-platform[bot] committed:
Creación workflow cloudbees close #59

cloudbees-platform[bot] committed:
Creación workflow cloudbees close #59

cloudbees-platform[bot] committed:
Creación workflow cloudbees close #59

cloudbees-platform[bot] committed:
workflow cloudbees close #59

cloudbees-platform[bot] committed:
Actualizo y elimino errores cloudbees.yaml close #59

cloudbees-platform[bot] committed:
Actualizo workflow cloudbees close #59

Christianlr committed:
Subida archivo configuración appveyor close #59

Christianlr committed:
Añado archivo de configuración appveyor close #59

Christianlr committed:
Añado configuración GHActions close #62

Christianlr committed:
Añador criterios y elección para herramientas CI close #58

Christianlr committed:
Modifico archivo iv.yaml con apartado CI close #63
añado apartado con los archivos de integración continua

Christianlr committed:
Añador apartado de elección herramientas CI close #64

Christianlr committed:
Modifico appveyor para docker close #65
añado configuración para poder lanzar el contenedor y ejecutar test

Christianlr committed:
Cambio de imagen en appveyor close #65

Christianlr committed:
Cambio la sentencia de ejecución en appveyor

Christianlr commented:
@JJ Listo para revisión

JJ commented:
¿No prefieres esperar a que lo revise alguien? Por ejemplo @pablobarto o @puchy22 (aunque no hayan salido aleatoriamente)

Christianlr commented:
Buenas @JJ, yo era por si quizás estaban algo ocupados ahora mismo. Aunque no pasa nada, espero alguna revisión de ellos :+1:

JJ commented:
Hombre, la verdad es que los que estaban asignados ya han hecho muchas revisiones y están bastante liados con sus propios objetivos, por eso he mencionado a los otros dos. Si no pueden, en un par de días o el lunes a más tardar te lo reviso.

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+## Elección de la herramienta de integración continua
+La integridad continua (CI) en el desarrollo de software implica la integración regular y automática de código con pruebas automatizadas frecuentes. Aporta beneficios al proyecto al detectar tempranamente errores, permitir entregas rápidas, mejorar la calidad del código, facilitar la colaboración en equipo, automatizar procesos de construcción y proporcionar retroalimentación inmediata. Por tanto ...
```
Creo que el término más correcto que "integridad continua" en español sería "integración continua".

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea :tada:

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+### Criterios de elección
+
+-   **Que sea fácil de obtener y duradera**: Son muchas las herramientas que se pueden encontrar en la web, pero muchas de ellas vienen ligadas a cortos periodos de prueba y a pagos mensuales. Se debe encontrar una herramienta que sea gratuita y además tenga o sea accesible el mayor tiempo posible.
+-   **Integración con GitHub**:  Esta herramienta que estamos buscando debe trabajar correctamente con GitHub y no debe tener problemas de integración con nuestro proyecto.
```
Este criterio creo que podría no tener mucho sentido, ya que si no es compatible con GitHub no sirve y no tiene sentido ni tenerlo en cuenta.

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea  :tada:

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+- [Semaphore CI](https://semaphoreci.com/): Proporciona un enfoque simple y eficiente para la CI/CD, admitiendo múltiples lenguajes y entornos de ejecución. Ofrece planes gratuitos y de pago con características adicionales.
+- [CloudBees](https://cloudbees.io/): Ofrece integración y orquestación continua para equipos ágiles, con soporte para pipelines complejas y despliegues continuos. Es una solución empresarial con opciones de pago.
+- [TeamCityCI](https://www.jetbrains.com/es-es/teamcity/): Herramienta de CI/CD potente y altamente personalizable, con escalabilidad y compatibilidad con diferentes tecnologías. Tiene versiones gratuita y de pago para empresas y equipos grandes.
+- [AppVeyor](https://www.appveyor.com/): Compatible con el proyecto y permite una configuración eficiente. Ofrece un plan gratuito con opciones de pago para características avanzadas.
```
Podrías entrar un poco más en detalle que diferencia a AppVeyor del resto ya que ha sido la escogida finalmente.

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea :tada:

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+
+### En cuanto a las versiones...
+
+Las versiones que van a ser testeadas son las que encontramos en la página de [NodeJS](https://nodejs.org/en). Estas se basan en la versión current y las LTS. La principal diferencia que hay entre ellas es el soporte, siendo las versiones LTS las más . Por otro lado, la version "current", es decir, la actual, es aquella versión
```
Este párrafo está sin terminar.

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea :tada:

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+
+Las versiones que van a ser testeadas son las que encontramos en la página de [NodeJS](https://nodejs.org/en). Estas se basan en la versión current y las LTS. La principal diferencia que hay entre ellas es el soporte, siendo las versiones LTS las más . Por otro lado, la version "current", es decir, la actual, es aquella versión
+
+El principal motivo para el testeo de las versiones es que se detecten imcompatibilidades del proyecto en otras versiones que puedan ser usadas y así garantizar que el proyecto es compatible en las distintas versiones estables y de producción, como las LTS, junto con las versiones actuales, sus características y mejoras.
```
incompatibilidades*

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea :tada:

puchy22 commented on .github/workflows/ci_test.yml:
```diff
@@ -0,0 +1,28 @@
+  push:
+
+jobs:
+  build:
```
Más que "build" creo que el nombre correcto para el trabajo sería "test" o algo similar.

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/dde8f3b4bbc14581ba2db73ddbd797cafaf56863 :tada:

puchy22 commented on docs/eleccion_herramienta_ci.md:
```diff
@@ -0,0 +1,30 @@
+
+### Herramientas consideradas
+
+Hay gran variedad de herramientas disponibles para aportar CI a un proyecto. Principalmente debemos tener en cuenta el tamaño del proyecto que se desarrolla y, en cuanto a la herramienta, si es fácil de obtener (si está ligada a una suscripción o periodo corto de prueba). He tenido en cuenta aquellos criterios mencionados en [G2](https://www.g2.com/) para la elección. Se tienen en cuenta las sigu ...
```
Ya que hablas que has tenido en cuenta [G2](https://www.g2.com/), creo que sería interesante que comentaras más particularmente qué criterio has usado en la consideración de algunas de las herramientas comentadas posteriormente.

Christianlr replied:
Arreglado en https://github.com/Christianlr/MIBarberSchedule/pull/61/commits/e746babeeac9652206646ccfbec86b684ab8b5ea :tada:

puchy22 reviewed (changes requested):
Feliz año @Christianlr :snowman_with_snow: , he revisado el objetivo y por lo general muy buen trabajo. He visto que has tenido en cuenta gran variedad de herramientas y has probado distintas versiones de node y distintos entornos. Revisa las anotaciones que te he dejado y a ver como avanza, suerte :four_leaf_clover:.

Christianlr committed:
Arreglo términos y explicaciones close #66
expreso con mayor claridad puntos mencionados en la elección de herramientas CI

Christianlr committed:
Cambio expresión en archivo con mayor contexto close #67
Se quieren lanzar test, por lo que no tiene sentido tener 'build'

Christianlr commented:
Buenas @puchy22 y feliz año nuevo! :snowman_with_snow:  Muchas gracias por la revisión tan rápida que has hecho. Estás que no paras! He arreglado todos aquellos comentarios que has hecho. Cuando puedas, revísalo a ver que te parece! :wink: :muscle:

Christianlr committed:
Elimino archivos no relevantes close #68

puchy22 reviewed (approved):
@Christianlr Por mí lo veo perfecto, con los cambios ha quedado mucho más completo :+1:

Christianlr commented:
@JJ Pues supongo que ahora si, cuando ya puedas, está listo para revisión (por cierto feliz año para ti también! 🙏 )

JJ commented on appveyor.yml:
```diff
@@ -0,0 +1,11 @@
+  - docker version
+
+build_script:
+  - docker build -t christianlr/mibarberschedule .
```
¿Por qué necesitas construirlo? ¿No te lo puedes bajar directamente del repo?

JJ commented on appveyor.yml:
```diff
@@ -0,0 +1,11 @@
+image: Ubuntu
+
+install:
+  - docker version
```
Esto en realidad no hace nada...

JJ reviewed (approved):
Queda el misterio de qué pasó con cloudbees... Y la configuración de AppVeyor, pero no son obstáculos para superar el objetivo.
